determined-ai/determined
# fix: det slot for no-permissions RBAC users

## Description

When RBAC enabled users with no task viewing permissions run `det slot` or `det slot list`,  task name should not be visible in the slot summary and should be replaced with a message indicating "No permissions". (As long as the task is a determined task) Task name is no longer listed as `Non-determined task`.


## Test Plan

To test the CLI display with `det slot list`:

- Spin up a determined EE cluster with RBAC enabled.
- Run `det user login admin` and `det rbac my-permissions` to ensure that `admin` has all necessary permissions to view and create Notebooks, Tensorboards, Shells, and Commands.
- Run `det shell start`
     - Run `exit` to exit out of the determined shell if it opens up in your terminal.
- Run `det slot list` to ensure that you can see shell allocation and all of its details while logged in as `admin`.
- Now, run `det user login determined` and `det rbac my-permissions` to ensure that your determined user has no RBAC permissions (if it does, create a new user with `admin` and log into that user, as newly created users should start out with no RBAC permissions).
- Run `det slot list` while logged in as user with no permissions to ensure that task name is obfuscated in slot summary for the shell task while logged in as user with no task viewing permissions. (Should give a message indicating that the user has no permissions).

## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket

DET-9595
